// store/catalog/mock.ts
export const ponies = [
  {
    id: "1",
    name: "Искорка",
    color: "фиолетовый",
    kind: "Единорог",
    price: 19.99,
    isNew: false
  },
  {
    id: "2",
    name: "Эпплджек",
    color: "оранжевый",
    kind: "Пони",
    price: 1.99,
    isNew: false
  },
  {
    id: "3",
    name: "Рэйнбоу Дэш",
    color: "голубой",
    kind: "Пегас",
    price: 22.00,
    isNew: true
  },
  {
    id: "4",
    name: "Рэрити",
    color: "фиолетовый",
    kind: "Единорог",
    price: 2.00,
    isNew: false
  },
  {
    id: "5",
    name: "Флаттершай",
    color: "розовый",
    kind: "Пегас",
    price: 21.0,
    isNew: true
  },
  {
    id: "6",
    name: "Пинки Пай",
    color: "розовый",
    kind: "Пони",
    price: 10,
    isNew: true
  },
  {
    id: "7",
    name: "Принцесса Селестия",
    color: "белый",
    kind: "Принцесса пони",
    price: 19.99,
    isNew: false
  },
  {
    id: "8",
    name: "Принцесса Луна",
    color: "синий",
    kind: "Принцесса пони",
    price: 19.99,
    isNew: false
  },
  {
    id: "9",
    name: "Принцесса Каденс",
    color: "розовый",
    kind: "Принцесса пони",
    price: 19.99,
    isNew: true
  },
  {
    id: "10",
    name: "Принцесса Скайла",
    color: "розовый",
    kind: "Принцесса пони",
    price: 19.99,
    isNew: true
  },
  {
    id: "11",
    name: "Шайнинг Армор",
    color: "белый",
    kind: "Единорог",
    price: 47,
    isNew: false
  },
  {
    id: "12",
    name: "Большой Макинтош",
    color: "красный",
    kind: "Жеребец",
    price: 10,
    isNew: true
  },
  {
    id: "13",
    name: "Бабуля Смит",
    color: "черный",
    kind: "Единорог",
    price: 30,
    isNew: false
  },
  {
    id: "14",
    name: "Зекора",
    color: "белый",
    kind: "Зебра",
    price: 17.54,
    isNew: false
  },
  {
    id: "15",
    name: "Чирили",
    color: "вишневый",
    kind: "Пони",
    price: 19.99,
    isNew: true
  },
  {
    id: "16",
    name: "Дерпи (Дитзи)",
    color: "белый",
    kind: "Пегас",
    price: 110,
    isNew: true
  },
  {
    id: "17",
    name: "Лайтнинг Даст",
    color: "бирюзовый",
    kind: "Пегас",
    price: 21.0,
    isNew: false
  },
  {
    id: "18",
    name: "Королева Крисалис",
    color: "зеленый",
    kind: "Аликорн",
    price: 10,
    isNew: true
  },
  {
    id: "19",
    name: "Король Сомбра",
    color: "черный",
    kind: "Единорог",
    price: 25,
    isNew: false
  },
  {
    id: "20",
    name: "Свити Белль",
    color: "белый",
    kind: "Единорог",
    price: 19.99,
    isNew: false
  }
];
