// store/catalog/reducer.ts
import { Reducer } from "redux";
import { CatalogState, CatalogTypes } from "store/catalog/types";

const initialState: CatalogState = {
  items: [
    {
      id: "1",
      name: "Ибрагима",
      color: "черный",
      kind: "Единорог",
      price: 19.99,
      isNew: false
    },
    {
      id: "2",
      name: "Андрей",
      color: "белый",
      kind: "Пони",
      price: 19.99,
      isNew: false
    },
    {
      id: "3",
      name: "Кудусай",
      color: "зеленый",
      kind: "Японский единорог",
      price: 19.99,
      isNew: true
    },
    {
      id: "4",
      name: "Лохантаэстамилока",
      color: "зеленый",
      kind: "Японский единорог",
      price: 19.99,
      isNew: true
    },
    {
      id: "5",
      name: "Кудусай",
      color: "зеленый",
      kind: "Японский единорог",
      price: 21.00,
      isNew: true
    },
    {
      id: "6",
      name: "Лохантаэстамилока",
      color: "зеленый",
      kind: "Японский единорог",
      price: 10,
      isNew: true
    }
  ],
  isShowFilter: false
};

const catalogReducer: Reducer<CatalogState> = (
  state = initialState,
  action
): CatalogState => {
  switch (action.type) {
    case CatalogTypes.SHOW_FILTER: {
      return { ...state, isShowFilter: true };
    }
    case CatalogTypes.CLOSE_FILTER: {
      return { ...state, isShowFilter: false };
    }
    default:
      return state;
  }
};

export default catalogReducer;
